extensions/amp-story/0.1/system-layer.js

```diff
@@ -146,14 +143,15 @@ export class SystemLayer {
 
     this.addEventHandlers_();
 
-    const systemLayerButtonsShown = this.stateService_
-        .getState(StateType.SYSTEM_LAYER_BUTTONS_SHOWN);
-
-    // TODO(newmuis): Observe this value.
-    if (!systemLayerButtonsShown.getValue() &&
-        !systemLayerButtonsShown.isModifiable()) {
-      this.root_.classList.add('i-amphtml-story-ui-no-buttons');
-    }
+    // The icon won't flash since this promise should be resolved instantly, but
+    // it is necessary to prevent any race condition.
```

Comment:
To be sure (and, perhaps, to future-proof this and make it clearer to others what's happening), should we return a promise here and block on it in AmpStory's layoutCallback?

Reply:
This was getting too complex for this PR as some code running right after assume the SystemLayer is built.
I was not super happy with that either, so I just introduced a way to retrieve the service synchronously instead.

Could you have another look? Thanks!